### ISSUE TITLE
Improved UX for keyboard and screen reader users through change in DOM order

### ISSUE THREAD

**Issue by Malvoz** [Leaflet/Leaflet#7479]
The DOM order of elements affects the order in which screen readers announce content, and also the tab order.

I believe that the map controls should be available/presented to the user before other author provided content such as markers and popups. In other words: `.leaflet-control-container` should come before `.leaflet-map-pane` in the DOM order.

This would reduce confusion and allow blind users to form a consistent mental model of the content. It also creates a more consistent and natural tab order, even across different maps (perhaps by different authors) with varying content.

Relates to <span title="Web Content Accessibility Guidelines">WCAG</span> 2.1 Success Criterion [2.4.3 Focus Order](https://www.w3.org/TR/WCAG21/#focus-order) and [3.2.3 Consistent Navigation](https://www.w3.org/TR/WCAG21/#consistent-navigation).

<hr>

@stevevance raised a similar concern in https://github.com/Leaflet/Leaflet/issues/3210 (citing the relevant parts):
> I was testing my website with a person who is completely blind and uses the VoiceOver (VO) function on an iPad. I was observing her as she browsed the entire webpage (through a tapping function in VO that mimics tabbing between elements*) and came across the Leaflet map.

> ## Expectations

> The next tab should land on the zoom controls, layer controls, and any other controls, so the user can understand what functions are available to control the map.

> The next tab should land on any marker that has a popup and the contents of the popup should be read.

**Comment by Malvoz** [comment]
> I believe that the map controls should be available/presented to the user before other author provided content such as markers and popups.

While I think this would be an improvement (especially when there are a lot of markers), the problem with this approach is that it causes a disruptive tab stop between panning and interacting with the markers.

Below is a description of 3 different methods that I think should be considered.

## A. Current scenario

The current scenario in a map with say 100 markers:

1. <kbd>Tab</kbd> to map
2.  <kbd>Tab</kbd> through 100 markers
3. <kbd>Tab</kbd> to controls

In this scenario, it would take more than 100 tab presses to reach the controls, to go into fullscreen for example. It's also harder for users relying on screen readers to detect the available controls.

## B. Changing DOM order (https://github.com/Leaflet/Leaflet/pull/7796)

If the DOM order of the map pane and controls is changed, the scenario would be:

1. <kbd>Tab</kbd> to map
2. <kbd>Tab</kbd> through a few controls
3. <kbd>Tab</kbd> to markers

Now it's easy for keyboard users to use the controls but will have to <kbd>Tab</kbd> between map -> controls -> markers and <kbd>Shift</kbd> + <kbd>Tab</kbd> between markers -> controls -> map when they only want to pan and interact with markers, i.e. they're forced to tab past the controls each time.

## C. Arrow key navigation for focusable elements in `leaflet-map-pane`

The optimal solution (that I can think of) would have to be something more like how e.g. Google's [marker accessibility](https://developers.google.com/maps/documentation/javascript/examples/marker-accessibility) works - markers are navigable using <kbd>arrow</kbd> keys.

This scenario would be (without changing current DOM order):

1. <kbd>Tab</kbd> to map
2. <kbd>Tab</kbd> to markers (now use <kbd>arrow</kbd> keys if you want to navigate between markers)
3. <kbd>Tab</kbd> to controls<br>
  There are potentially 2 ways to achieve this:
    1. Implement something like the [ARIA Grid pattern](https://www.w3.org/TR/wai-aria-practices-1.1/#grid)
     (the more complex option)
    2. Set the `focusgroup` attribute on `leaflet-map-pane`, done!
     (well, that's in theory, [`focusgroup` is a new proposal](https://github.com/MicrosoftEdge/MSEdgeExplainers/blob/main/Focusgroup/explainer.md) by Microsoft, which would need to be [polyfilled](https://github.com/MicrosoftEdge/MSEdgeExplainers/blob/main/Focusgroup/focusgroup_polyfill.js))

**Comment by Falke-Design** [comment]
I think solution "C" would be good. But I don't think that the other maintainers allows the polyfill for `focusgroup`. Btw. with focusgroup it looks like [focusegroup='extend'](https://github.com/MicrosoftEdge/MSEdgeExplainers/blob/main/Focusgroup/explainer.md#632-extend---extending-the-focusgroup) would be needed too.

So we need to use the ARIA Grid pattern.
